bin/citrea/src/main.rs

```diff
@@ -150,7 +150,11 @@ async fn main() -> Result<(), anyhow::Error> {
 
     let mut network = args.network.into();
     if args.dev {
-        network = Network::Nightly;
+        network = if cfg!(feature = "testing") && network == Network::Regtest {
```

Comment:
Maybe just use `--network regtest` when needed. I don't think it's going to be convenient to always use regtest when running tests. What do you think?